bundles/org.openhab.binding.somfymylink/src/main/java/org/openhab/binding/somfymylink/internal/handler/SomfyMyLinkBridgeHandler.java

```diff
@@ -0,0 +1,478 @@
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerService;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.StateOption;
+import org.openhab.binding.somfymylink.internal.SomfyMyLinkBindingConstants;
+import org.openhab.binding.somfymylink.internal.config.SomfyMyLinkConfiguration;
+import org.openhab.binding.somfymylink.internal.discovery.SomfyMyLinkDeviceDiscoveryService;
+import org.openhab.binding.somfymylink.internal.model.SomfyMyLinkCommandBase;
+import org.openhab.binding.somfymylink.internal.model.SomfyMyLinkCommandSceneList;
+import org.openhab.binding.somfymylink.internal.model.SomfyMyLinkCommandSceneSet;
+import org.openhab.binding.somfymylink.internal.model.SomfyMyLinkCommandShadeDown;
+import org.openhab.binding.somfymylink.internal.model.SomfyMyLinkCommandShadeList;
+import org.openhab.binding.somfymylink.internal.model.SomfyMyLinkCommandShadePing;
+import org.openhab.binding.somfymylink.internal.model.SomfyMyLinkCommandShadeStop;
+import org.openhab.binding.somfymylink.internal.model.SomfyMyLinkCommandShadeUp;
+import org.openhab.binding.somfymylink.internal.model.SomfyMyLinkErrorResponse;
+import org.openhab.binding.somfymylink.internal.model.SomfyMyLinkPingResponse;
+import org.openhab.binding.somfymylink.internal.model.SomfyMyLinkResponseBase;
+import org.openhab.binding.somfymylink.internal.model.SomfyMyLinkScene;
+import org.openhab.binding.somfymylink.internal.model.SomfyMyLinkScenesResponse;
+import org.openhab.binding.somfymylink.internal.model.SomfyMyLinkShade;
+import org.openhab.binding.somfymylink.internal.model.SomfyMyLinkShadesResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/**
+ * The {@link SomfyMyLinkBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Chris Johnson - Initial contribution
+ */
+@NonNullByDefault
+public class SomfyMyLinkBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SomfyMyLinkBridgeHandler.class);
+    private static final int HEARTBEAT_MINUTES = 2;
+    private static final int MYLINK_PORT = 44100;
+    private static final int MYLINK_DEFAULT_TIMEOUT = 5000;
+    private static final int CONNECTION_DELAY = 1000;
+    private static final SomfyMyLinkShade[] EMPTY_SHADE_LIST = new SomfyMyLinkShade[0];
+
+    private @Nullable SomfyMyLinkConfiguration config;
```

Comment:
A common way to eliminate the need to null checking your config is to initialize the config to default value.
```suggestion
    private SomfyMyLinkConfiguration config = new SomfyMyLinkConfiguration();
```

Reply:
done